### PR TITLE
EES-2419 - notify release approvers

### PR DESCRIPTION
**Pull request by luke-h1**
This PR:
* notifies release approvers via notify that a release has moved into higher review. 
* Renames a few files that seemed to have erroneous spacing in them i.e. `MetaController .cs` -> `MetaController.cs`
* Makes amends to frontend + backend unit & UI tests where needed
* Adds a `NotifyReleaseApprovers` column to `ReleaseStatus` for auditing/tracking purposes.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IEmailTemplateService.cs [hunk 1]
```diff
@@ -23,5 +22,7 @@ public interface IEmailTemplateService
             string email,
             Release release,
             ReleaseRole role);
+
+        public Either<ActionResult, Unit> SendReleaseApproverEmail(string email, Release release);
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IEmailTemplateService.cs:26]: As it's an interface, every method declared in it is public by default, so no need to include the `public` keyword here :slightly_smiling_face:

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs [hunk 2]
```diff
@@ -1628,6 +1689,269 @@ await using (var context = InMemoryApplicationDbContext(contextId))
...
+                User = new User { Id = Guid.NewGuid(), Email = "test2@test.com"},
+                Release = release,
+                Role = ReleaseRole.Approver,
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                await context.AddRangeAsync(release, userReleaseRole1, userReleaseRole2);
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs:1900]: You could probably remove this saving of these ReleaseRoles to the databases entirely from the test.  Looks like it's a Mock of the IUserReleaseRoleService that's returning them, and so it won;t need to be looking in any DbContexts for data.

Alternatively you could swap out the Mock IUserReleaseRoleService for a real one and pass it the db context you've populated, in which case yes, you would need to persist the data here first :slightly_smiling_face:

File: src/GovUk.Education.ExploreEducationStatistics.Admin/appsettings.Development.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
   "NotifyReleaseRoleTemplateId": "change-me",
   "NotifyPreReleaseTemplateId": "change-me",
   "NotifyContributorTemplateId": "change-me",
+  "NotifyReleaseApproversTemplateId": "change-me",
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/appsettings.Development.json:9]: You'll also need to make associated changes in `infrastructure/template.json` too - see *case-insensitive* examples of `NotifyReleaseRoleTemplateId` in the `infrastructure` folder.

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs [hunk 4]
```diff
@@ -1628,6 +1689,269 @@ await using (var context = InMemoryApplicationDbContext(contextId))
...
+                contentService.Setup(mock => mock.GetContentBlocks<HtmlBlock>(release.Id))
+                    .ReturnsAsync(new List<HtmlBlock>());
+
+                preReleaseUserService.Setup(mock =>
+                        mock.SendPreReleaseUserInviteEmails(release.Id))
+                    .ReturnsAsync(Unit.Instance);
+
+                userReleaseRoleService.Setup(mock =>
+                        mock.ListUserReleaseRolesByPublication(ReleaseRole.Approver, release.Publication.Id))
+                    .ReturnsAsync(new List<UserReleaseRole>{ userReleaseRole1, userReleaseRole2 });
```
[the remark this answers is not in the document]
Comment (luke-h1) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs:1755]: Sweet, replaced all instances where I've included `userReleaseRole1` & `userReleaseRole2`

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs [hunk 5]
```diff
@@ -1628,6 +1689,269 @@ await using (var context = InMemoryApplicationDbContext(contextId))
...
+                        release.Id, new ReleaseStatusCreateViewModel
+                        {
+                            PublishMethod = PublishMethod.Scheduled,
+                            PublishScheduled = "2051-06-30",
+                            ApprovalStatus = ReleaseApprovalStatus.HigherLevelReview,
+                            LatestInternalReleaseNote = "Test internal note",
+                            NextReleaseDate = new PartialDate { Month = "12", Year = "2077" }
+                        });
+
+                VerifyAllMocks(contentService,
```
[the remark this answers is not in the document]
Comment (luke-h1) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseApprovalServiceTests.cs:1855]: Good catch, ended up getting rid of a few mocks that weren't needed for this particular test case

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**